### PR TITLE
fix: addon path generating on windows

### PR DESCRIPTION
**Pull request by dannyhw**
Issue: #645 

## How to test

Please explain how to test your changes and consider the following questions

- Does this need a new example in examples/expo-example?
- Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: packages/react-native/scripts/common.js [hunk 1]
```diff
@@ -69,7 +69,7 @@ function resolveAddonFile(addon, file, extensions = ['js', 'mjs', 'ts']) {
 
   for (const ext of extensions) {
     try {
-      const filePath = path.join(addon, `${file}.${ext}`);
+      const filePath = `${addon}/${file}.${ext}`;
```
Comment (trajano) [inline packages/react-native/scripts/common.js:72]: looks janky but I guess it should work.

**Comment by trajano** [comment]
Thanks looks like it exported correctly can you add in the fix for the withSTorybook.d.ts

**Comment by dannyhw** [comment]
@trajano added a fix in 8.4.2-alpha.1 for withStorybook can you give it a try?

File: packages/react-native/scripts/common.js (hunk 1 above)
Comment (dannyhw) [reply packages/react-native/scripts/common.js:72]: 😅 yeah I know but atleast I know what its doing haha

**Comment by trajano** [comment]
oof no it's not backward compatible but it is cleaner.  I really hate the nuances of the default exports

**Comment by trajano** [comment]
ChatGPTs's answer

Yes, Lord Archimedes, you can export the function both ways by doing the following:

```typescript
export function withStorybook() {
  // function implementation
}

export default withStorybook;
```

This allows `withStorybook` to be imported as either a named or default export.

**Comment by dannyhw** [comment]
@trajano unfortunately I couldn't get that to work but I used this and I think its working now
```
export { withStorybook };

module.exports = withStorybook;

module.exports.withStorybook = withStorybook;
```

8.4.2-alpha.2

**Comment by trajano** [comment]
weird that you have to go through that. you use tsup the same way I do too.

**Comment by dannyhw** [comment]
if you use it like this you get the type info properly too

```
const { withStorybook } = require('@storybook/react-native/metro/withStorybook');

module.exports = withStorybook(defaultConfig);
```

**Comment by trajano** [comment]
you know the weird part, you have two .js files one in `metro` and the other in `dist/metro`

**Comment by trajano** [comment]
This types properly
```
const {
  withStorybook,
} = require('@storybook/react-native/metro/withStorybook');
```
But is not backwards compatible.

The way I test is to go to visual studio code on my metro.config.js and check if it type checks correctly

**Comment by trajano** [comment]
Is it working for you though @dannyhw ?

**Comment by dannyhw** [comment]
It works but the default export doesnt show the types for some reason.

And the extra file is actually a fallback for file based resolution for example when packageExports is off

**Comment by trajano** [comment]
Okay, so far it's working for me with the `export { withStorybook }`  style on both WebStorm and VSCode.  TSC doesn't do any typechecks from what I can tell even with `// @ts-check`.  I verified by intentionally putting in `enabled: "yes"` which validates on TSC but not VSCode and WebStorm

**Comment by dannyhw** [comment]
im going to just push the path fix for now this withstorybook thing is proving to be more complex than expected

**Comment by dannyhw** [comment]
i will create another pr to try fix the with storybook thing soon

**Pull request merged**